homeassistant/components/homekit_controller/connection.py

```diff
@@ -247,25 +250,30 @@ def async_load_platforms(self):
             _LOGGER.debug("HomeKit connection not polling any characteristics.")
             return
 
-        _LOGGER.debug("Starting HomeKit controller update")
-
-        try:
-            new_values_dict = await self.get_characteristics(
-                self.pollable_characteristics
-            )
-        except AccessoryNotFoundError:
-            # Not only did the connection fail, but also the accessory is not
-            # visible on the network.
-            self.async_set_unavailable()
-            return
-        except (AccessoryDisconnectedError, EncryptionError):
-            # Temporary connection failure. Device is still available but our
-            # connection was dropped.
+        if self._polling_lock.locked():
+            _LOGGER.warning("HomeKit controller update skipped as previous poll still in flight")
```

Comment:
Do you think affected users will complain about the warning message?

Is it critical in the long term for the user to fix this problem, so that the user must be made aware?

Reply:
Good question. This logging message is supposed to only fire when the system is under so much load an operation that normally takes less than 1s is taking over a minute. The most likely causes for that are going to be bad devices, bad wifi or thread pool contention. So I'm hopeful most users won't see this and its a sign that their installation is deeply borked, so they will appreciate a clue. In that case I could see a different error message that said something like

> HomeKit controller update skipped as previous poll still in flight - your accessory might not have a good connection to your network, your system may be unable to handle the number of accessories and integrations, or your configuration maybe experiencing threadpool contention issues.

But thats verbose.

A counter argument is that although HomeKit controller can exacerabate the situation (especially in the bad wifi case) its not neccesarily the root cause, so even if it's taking action to relieve pressure on the users system, maybe it shouldn't implicate itself with a warning level message.

I'm happy to change as you see fit.

Reply:
It sounds like we can keep it as warning. The alternative is to make it debug I think.

We could add a section to the docs, explaining the warning?

Reply:
I've been meaning to improve the homekit_controller docs so i've included this in there: https://github.com/home-assistant/home-assistant.io/pull/10153

Reply:
I would say that we should warn once per "occurrence". So if we hit 5 warnings in a row, we only warn the first. We would also add an info log when we are back to normal polling.

Reply:
Have added the logging as suggested